packages/orchestrator-ui-components/src/components/WfoWorkflowSteps/WfoWorkflowStepList/WfoWorkflowStepList.tsx

```diff
@@ -35,18 +48,50 @@ export const WfoWorkflowStepList = React.forwardRef(
         const initialStepListItems: StepListItem[] = steps.map((step) => ({
             step,
             isExpanded: false,
-        }));
+        })); // If the step is in the suspend state, we show the user input form
 
         const [stepListItems, setStepListItems] =
             useState(initialStepListItems);
 
-        useEffect(
-            () =>
-                setStepListItems((previousStepListItems) =>
-                    updateStepListItems(previousStepListItems, steps),
+        const persistStepListItemState = (
+            previousStepListItems: StepListItem[],
+            updatedSteps: Step[],
+            userInputForm?: InputForm,
+        ): StepListItem[] => {
+            const reversedSteps = [...updatedSteps].reverse();
+            const lastStepWithSuspendStatusStepId = reversedSteps.find(
+                (step) => step.status === StepStatus.SUSPEND,
+            )?.stepId;
+
+            return updatedSteps.map((step) => {
+                const previousStepListItem = previousStepListItems.find(
+                    (previousStepListItem) =>
+                        previousStepListItem.step.stepId === step.stepId,
+                );
+                return {
+                    step,
+                    isExpanded: previousStepListItem
```

Comment:
Not sure if it's possible to use something like:

isExpanded: previousStepListItem?.isExpanded ?? false

Reply:
It is and I will change it but is that a lot clearer?

Reply:
Not sure about clearer and what style you use in the GUI. I'm fine with both.